strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NpmArtifactGenerator.java

```diff
@@ -219,6 +220,8 @@ private Path buildPublishJson()
         {
             buildPackage();
         }
+        return packagePath;
+        /* TODO @sbespalov - do we need this ? publish.json is then used as an artifact but NpmArtifactCoordinates.getArtifactFileName resolves it to package.json
```

Comment:
@sbespalov could you please take a look on this comment ?

Reply:
@ptirador seems I see what you mean, and probably there is a confusing here for real.
we need this `package.json` for `NpmArtifactController` (see `NpmArtifactControllerTest`), so when you need to publish the artifact then you should use it.
but from the other side you noticed correctly that in case of `ArtifactManagementTestExecutionListener` we should use just `NpmArtifactGenerator.buildPackage()`.
so finally we shouldn't comment this, but `NpmArtifactGenerator.generateArtifact` methods should use `buildPackage()` instead of `buildPublishJson()`.

Reply:
I think that comment this is making fail the `NpmArtifactControllerTest` test. So what do we do about this?

Reply:
yeah, seems like another issue :(

> I managed it to work with this, see gist.

I suggest to resolve it other way, in this we can call still call the `buildPublishJson()` but return the `getPackagePath()` as a result of `generateArtifact` methods. This will generate the `publish.json` as well and you can get it using `packagePath.resolveSbilling("package.json")` within `NpmArtifactControllerTest`.

Reply:
@ptirador also regarding the `NpmArtifactControllerTest`, can you also drop `//Publish` part within `testViewPackage()`? we don't need it since the injected test artifact already deployed.

and for `testPackageCommonFlow()` you shouldn't set the `storageId` and `repositoryId` for injected artifact because in this case we will publish it manually within test method.

Reply:
@sbespalov About the `buildPublishJson()` issue, did you mean this?
See [commit](https://github.com/strongbox/strongbox/pull/1416/commits/9e705f6f020c3750229a5ada6ae2d84bde36fc2e).

The thing is that this is provoking fail again in `NpmArtifactControllerTest.testViewPackage` method:
See [Jenkins Job](https://jenkins.carlspring.org/blue/organizations/jenkins/strongbox%2Fbuilds%2Fstrongbox/detail/PR-1416/7/tests).

Reply:
@ptirador I've commented [here](https://github.com/strongbox/strongbox/pull/1416/commits/9e705f6f020c3750229a5ada6ae2d84bde36fc2e#r311854530)

Reply:
Okay, but the error that I said would still happen with this change. Should I disable this failed test and create an issue to fix it?

Reply:
ok, let's proceed like this